### ISSUE TITLE
Axios failing to use validateStatus.

### ISSUE THREAD

**Issue by markskayff** [axios/axios#5122]
### Describe the issue

The `validateStatus: function()` config option is supposed to middleware the response, and if validateStatus evaluates to **false** then the promise will be rejected, aka, an Excepton is thrown and we are in a catch block.

For some reason, I see the validaStatus block is evaluated (I have output on the code), but the Exception is not being thrown. This means, the then() block continues to execute, causing all sort of bugs in the code, as it should not execute at all.

I'm using and testing with Firefox 101.
Chrome reports the same bug.

Interesting thing is that this is not happening in my development setup. It's happening on a production machine.

### Example Code

```js
`axios.defaults.validateStatus = (status) => {                                                                                                                                                      
   //console.log("Validating status:", status >= 200 && status <= 304);                                                                                                                         
   return status >= 200 && status <= 304; // default                                                                                                                                            
};`
```

### Expected behavior

The expcted behaviour would be Axios throwing an exception and code falling in the rejected catch blocks.
Instead execution continues on the then block, swiping everything in its path.

### Axios Version

1.1.2

### Browser

Firefox

### Browser Version

101

### Node.js Version

16.17.1

### OS

Ubuntu 22.04.1 LTS

### Additional Library Versions

```bash
"react": "^17.0.0"
"redux": "^4.1.2"
```

**Comment by markskayff** [comment]
Fixed. It was an internal bug, due to some code in an interceptor.

**Comment by BuffaloBuffalo** [comment]
@markskayff, is there any indication in what version this was fixed in (I'm guessing 1.1.3?) or what PR addressed it? Alternatively, how long it existed? Is it present in the old 0.27.x branch?
